src/main/java/spoon/javadoc/internal/Javadoc.java

```diff
@@ -0,0 +1,222 @@
+		* at the start of the line.
+		*/
+	public String toText() {
+		StringBuilder sb = new StringBuilder();
+		if (!description.isEmpty()) {
+			sb.append(description.toText());
+			sb.append(System.lineSeparator());
+		}
+		if (!blockTags.isEmpty()) {
+			sb.append(System.lineSeparator());
+		}
+		blockTags.forEach(
+			bt -> {
+				sb.append(bt.toText());
+				sb.append(System.lineSeparator());
+			});
+		return sb.toString();
+	}
+
+	public JavadocDescription getDescription() {
+		return description;
+	}
+
+	/** @return the current List of associated JavadocBlockTags */
+	public List<JavadocBlockTag> getBlockTags() {
+		return this.blockTags;
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || getClass() != o.getClass()) {
+			return false;
+		}
+
+		Javadoc document = (Javadoc) o;
+
+		return description.equals(document.description) && blockTags.equals(document.blockTags);
+	}
+
+	@Override
+	public int hashCode() {
+		int result = description.hashCode();
+		result = 31 * result + blockTags.hashCode();
+		return result;
+	}
+
+	@Override
+	public String toString() {
+		return "Javadoc{" + "description=" + description + ", blockTags=" + blockTags + '}';
+	}
+
+	private static String BLOCK_TAG_PREFIX = "@";
+	private static Pattern BLOCK_PATTERN =
+			Pattern.compile("^\\s*" + BLOCK_TAG_PREFIX, Pattern.MULTILINE);
+
+	private static final String EOL = System.lineSeparator();
+
+	/** parse the description part (before tags) of a Javadoc */
+	public static JavadocDescription parseText(String text) {
+		JavadocDescription instance = new JavadocDescription();
+		int index = 0;
+		Pair<Integer, Integer> nextInlineTagPos;
+		while ((nextInlineTagPos = indexOfNextInlineTag(text, index)) != null) {
+			if (nextInlineTagPos.a != index) {
+			instance.addElement(new JavadocSnippet(text.substring(index, nextInlineTagPos.a)));
+			}
+			instance.addElement(
+				JavadocInlineTag.fromText(text.substring(nextInlineTagPos.a, nextInlineTagPos.b + 1)));
+			index = nextInlineTagPos.b + 1;
+		}
+		if (index < text.length()) {
+			instance.addElement(new JavadocSnippet(text.substring(index)));
+		}
+		return instance;
+	}
+
+	private static Pair<Integer, Integer> indexOfNextInlineTag(String text, int start) {
+		int index = text.indexOf("{@", start);
+		if (index == -1) {
+			return null;
+		}
+		// we are interested only in complete inline tags
+		int closeIndex = text.indexOf("}", index);
+		if (closeIndex == -1) {
+			return null;
+		}
+		return new Pair<>(index, closeIndex);
+	}
+
+	/** parses the Javadoc content (description + tags) */
+	public static Javadoc parse(String commentContent) {
+		List<String> cleanLines;
+
+		// In JavaParse, ,they had their own version of cleaning
+		// not backward compatible with Spoon's one
+		// so we keep Spoon's one
+		// cleanLines = cleanLines(commentContent);
+		cleanLines = Arrays.asList(commentContent.split(EOL));
```

Comment:
> I took care of not writing anything OS-dependent.

The input string actually always contains `\n`. But constant `EOL` on this line is OS dependent EOL == System.lineSeparator();

There is more uses of `System.lineSeparator()`, I suggest to replace them by "\n", so we keep already established concept that lines in Javadoc values in Spoon are always separated by `\n` ... as long as they are in form of String values of the spoon model. Once they are printed, the EOL vary depending on OS or Spoon configuration.

Reply:
Here you mix two things: the line separator of the input source (here), and the line separator in `getContent()`

`getContent()` uses `\n`, see `ret.append(CtComment.LINE_SEPARATOR)`

Reply:
no. I speak about the line separator of the input source (here). This method is called here 
```java
javadoc = Javadoc.parse(cleanComment(content));
```
and `cleanComment` always returns `\n` as line separator, therefore input of this `parse` method always has `\n` as line separator ... but `EOL` constant defined above is actually `\r\n` on Windows so it cannot work.